### DISCUSSION TITLE
Difference between rtl_433 and actual hardware

### DISCUSSION THREAD

**Discussion by Crsarmv7l** [merbanan/rtl_433#2642] [Q&A] [answered]
Posted a few times on here about recreating signals and whatnot. I came across an interesting occurrence and I am not sure why this is the case, only that it is.

I created a low pressure signal for PMV-107J that I am sending to rtl_433 from my Yardstick One. With a datarate of ~19200 rtl_433 sees and decodes the transmission (even multiple transmissions).

Testing on my wife's Toyota 4 Runner which uses the PMV-107J, the car trips the low pressure light with a datarate of 9600....yet rtl_433 does not see the transmission at that datarate and the car does not see the transmission at 19200.

Looking at a real signal in URH, each bit takes about ~13us. 1,000,000us in a sec/13us = ~76923 bit/s or 9615 byte/s

I understand that receivers tend to allow variance from the exact spec, but 19200 and 9600 are really far apart and I am wondering why?

**Comment by gdt** [comment, marked as answer]
I would suggest comparing the recordings (waterfall, not just decodes) from real TPMS vs your synthetic.   Twice the rate could be an error in creating it, in terms of some kind of machester encoding.    I suspect something is off.    Maybe you can post a link to a repo someplace that has your generation code?

**Reply by Crsarmv7l** [reply]
Definitely not off. It produces bit for bit exactly the same bitstring as any of the reference files when the same values are used.

Additionally  if that were the case I would expect there to be a difference between the repeats sent and the repeats decoded. I play 4 signals to rtl_433, I get 4 decodings (with correct values). I play one signal transmission to the actual car, and immediately get a light.

My suspicion is the bitstring is converted to bytes which are transmitted by rfcat (the Yardstick One does not use IQ) and there is a timing issue of some sort with that. Still doesn't make sense though.

I'll see if I can get captures of both a bit later tonight using values from a ref .cu8 so there is one real and two fake at the different rates, all with the same values

**Reply by Crsarmv7l** [reply]
You are a prince! 

My bitstring was correct and accurate to a ref file from the test files repo as generated, but I noticed something odd when I opened up a capture of my synthetic. There was what looked like Manchester before the signal, that was not in the generated bitstring.

eg:
Printed bitstring started something like 00000001111100

While the capture when examined looked something like 101010101000000001111100

After a few minutes I finally realized the cc1111 was inserting a preamble/syncword before the transmission, and since the bitbuffer/decoders look for certain lengths it wasn't seeing my transmission at 9600 (it was too long), apparently jacking up the speed to 19200 was enough to "minimize" the preamble/sync enough for the actual signal to be small enough to be decoded.

After throwing in d.setMdmSyncMode(0), there was no more preamble/sync and my synthetic was recognized at 9600 datarate in rtl_433 with no other changes!

It all makes sense now. lol

Thanks bud
